### PR TITLE
REV-171-create-saved-listings-tab

### PR DESCRIPTION
**Pull request by IbrahimAldereni-socuimtech**
### Jira URL

https://optimumpartners.atlassian.net/browse/REV-171
https://optimumpartners.atlassian.net/browse/REV-172

### Description

- get user saved users and listings

### Dependencies in Other PRs,Repo

- backend PR: https://github.com/abukhadraop-or/reverifi-api/pull/41

Screenshot from 2022-02-28 17-45-31
Screenshot from 2022-02-28 17-46-14
Screenshot from 2022-02-28 17-46-10
Screenshot from 2022-02-28 17-45-37

### CODE REVIEW

File: src/components/shared/Tabs/tabs.styles.js [hunk 1]
```diff
@@ -20,3 +22,10 @@ export const TabContent = styled.div`
     margin: 0;
   }
 `;
+
+export const Line = styled.hr`
```
[the remark this answers is not in the document]
Comment (mohdabed123) [reply src/components/shared/Tabs/tabs.styles.js:26]: OR Separator

File: src/components/shared/SaveAndShareButtons/index.jsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ import {
  * @param {String} userId            user id
  * @param {String} listingId         listing id
  * @param {Function} setShowModal    show modal function
- * @param {Boolean} small            small icons if true
+ * @param {Boolean} small            small icons if true *
```
Comment (mohdabed123) [inline src/components/shared/SaveAndShareButtons/index.jsx:24]:
```suggestion
 * @param {Boolean} small            small icons if true
```

File: src/components/saved-agents-listings/SavedListings/saved-listings.styles.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+export const IconContainer = styled.button`
+  background: none;
+  border: none;
+  cursor: pointer;
+  margin: 0 0.3rem;
+  padding: ${({ hover }) => (hover ? '0.35rem' : 0)};
+  position: relative;
+
+  &:hover {
+    background: ${({ hover }) => hover && colors.mercury};
```
Comment (mohdabed123) [inline src/components/saved-agents-listings/SavedListings/saved-listings.styles.js:42]:
```suggestion
    background: ${({ hoverable }) => hover && colors.mercury};
```

File: src/components/saved-agents-listings/SavedAgentsListingsWrapper/index.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+ *
+ * @return {JSX.Element}
+ */
+function SavedAgentsListingsWrapper() {
+  return (
+    <SavedAgentsListingsContainer>
+      <SavedText>Saved</SavedText>
+      <Tabs
+        tabsTitles={['Saved Listings', 'Saved Agents']}
+        tabsContent={[<SavedListings />, <SavedAgents />]}
```
Comment (mohdabed123) [inline src/components/saved-agents-listings/SavedAgentsListingsWrapper/index.jsx:21]: Can we put these in a const array of objects

File: src/components/saved-agents-listings/NoDataComponent/index.jsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ * @param {String} buttonText text to show on button
+ *
+ * @return {JSX.Element}
+ */
+function NoDataComponent({ image, dataFor, buttonText }) {
+  const navigate = useNavigate();
+  return (
+    <NoDataContainer>
+      <NoDataImage src={image} />
+      <h2>you haven&apos;t saved any {dataFor} yet</h2>
```
Comment (mohdabed123) [inline src/components/saved-agents-listings/NoDataComponent/index.jsx:21]:
```suggestion
      <NoDataImage src={image} />
      <h2>you haven't saved any {dataFor} yet</h2>
```

Comment (mohdabed123) [reply src/components/saved-agents-listings/NoDataComponent/index.jsx:21]: &apos is not part of the HTML standard

File: src/components/saved-agents-listings/NoDataComponent/index.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      <p>
+        Use the heart icon to like the {dataFor} you&apos;re interested in and
```
Comment (mohdabed123) [inline src/components/saved-agents-listings/NoDataComponent/index.jsx:23]:
```suggestion
        Use the heart icon to like the {dataFor} you are interested in and
```

File: src/components/saved-agents-listings/NoDataComponent/index.jsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * No Data Component
+ *
+ * @param {String} image  no data image
+ * @param {String} dataFor data for listing or agent
+ * @param {String} buttonText text to show on button
+ *
+ * @return {JSX.Element}
+ */
+function NoDataComponent({ image, dataFor, buttonText }) {
```
Comment (mohdabed123) [inline src/components/saved-agents-listings/NoDataComponent/index.jsx:16]:
```suggestion
function NoDataComponent({ image, agentData, buttonText }) {
```

File: src/components/saved-agents-listings/SavedAgents/index.jsx [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+ *
+ * @return {JSX.Element}
+ */
+function SavedAgents() {
+  const { isLoggedIn, userInfo: authInfo } = useUser();
+  const navigate = useNavigate();
+  const [users, setUsers] = useState([]);
+  const [pageNumber, setPageNumber] = useState(0);
+
+  const limit = 8;
```
Comment (mohdabed123) [inline src/components/saved-agents-listings/SavedAgents/index.jsx:38]:
```suggestion
  const PAGE_LIMIT = 8;
```

**Review by mohdabed123 — CHANGES_REQUESTED** [review]
Note: we should consider making a pagination shared component probably in a separate ticket.

File: src/components/saved-agents-listings/SavedAgentsListingsWrapper/index.jsx (hunk 4 above)
Comment (IbrahimAldereni-socuimtech) [reply src/components/saved-agents-listings/SavedAgentsListingsWrapper/index.jsx:21]: ibrahim aljaber work on that on REV-166

File: src/components/saved-agents-listings/NoDataComponent/index.jsx (hunk 7 above)
Comment (IbrahimAldereni-socuimtech) [reply src/components/saved-agents-listings/NoDataComponent/index.jsx:16]: thats can be user of listing

**Pull request merged**